### PR TITLE
Refs [TASK][36598] : Data Detail Screen

### PR DESCRIPTION
**Pull request by manhhung23399**
## Related Tickets
- [#Task 36598: Data Detail Screen](https://edu-redmine.sun-asterisk.vn/issues/36598)
## WHAT
- check favorite
- insert game to favorite
- bind image platform
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/gamevui/ui/PcFragment.kt [hunk 1]
```diff
@@ -67,12 +67,12 @@ class PcFragment : BaseFragment<FragmentPlatformBinding>() {
     }
 
     private fun clickItemPopular(game: Game) {
-        val action = HomeFragmentDirections.actionHomeFragmentToGameDetailFragment(game.id)
+        val action = HomeFragmentDirections.actionHomeFragmentToGameDetailFragment(game)
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/gamevui/ui/PcFragment.kt:70]: sao đang đẻ id lại truyền cả game thế, cho n nặng thêm ra

File: app/src/main/java/com/sun/gamevui/ui/GameDetailViewModel.kt [hunk 2]
```diff
@@ -36,4 +40,19 @@ class GameDetailViewModel(
             _screenshots.postValue(screenshotFromApi.results)
         }
     }
+
+    fun checkFavorite(id: Long) {
+        viewModelScope.launch(Dispatchers.IO + exceptionHandler) {
+            val gameFavorite = gameRepo.isFavorite(id)
+            _isFavorite.postValue(gameFavorite != null)
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/gamevui/ui/GameDetailViewModel.kt:47]: sao phải check != null thế. Với lại mình check base viewmodel tháy bạn đi cancel viewmodel scope, bản thân n tự lm cái đso r thì cần gì viết lại nữa nhỉ

File: app/src/main/java/com/sun/gamevui/ui/PcFragment.kt (hunk 1 above)
Comment (manhhung23399) [reply app/src/main/java/com/sun/gamevui/ui/PcFragment.kt:70]: e truyền để sang bên detail còn instert game đc game a

File: app/src/main/java/com/sun/gamevui/ui/GameDetailViewModel.kt (hunk 2 above)
Comment (manhhung23399) [reply app/src/main/java/com/sun/gamevui/ui/GameDetailViewModel.kt:47]: check null thì để xem nó có trong saved ko để hiển thị image cho đúng a
cái vụ viewmodel scope tự cancel thì e ko rõ e viết cho chắc ạ

**Pull request merged**